odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,254 @@
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator domain
+        attenuation : `element-like` structure of the shape
+            `dom.shape`, optional
+            Linear attenuation map for SPECT.
+        psf : `element-like` structure, optional
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        impl={'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `niftypy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `NiftyPy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and GPU
+        """
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
+                raise ValueError('NiftyRec back-end not available.')
+            if impl == 'niftyrec_gpu' and dom.shape != (128, 128, 128):
+                raise NotImplementedError('NiftyRec GPU back-end only '
+                                          'supports volume size '
+                                          '(128, 128, 128), got {}'
+                                          ''.format(dom.shape))
+            if impl == 'niftyrec_gpu':
+                self._use_gpu = True
+            elif impl == 'niftyrec':
+                self._use_gpu = False
+
+        self._geometry = geometry
+        det_nx, det_ny = geometry.det_grid.shape
+        startAngle = geometry.angles[0]
+        stopAngle = geometry.angles[-1]
+        N_proj = geometry.motion_grid.size
+
+        self._cameras = np.linspace(startAngle, stopAngle, N_proj,
+                                    dtype=np.float32).reshape((N_proj, 1))
+        if psf is not None:
+            self._psf = np.asarray(psf)
+        else:
+            self._psf = psf
+        if attenuation is not None:
+            self._attenuation = self.domain.element(attenuation)
+        else:
+            self._attenuation = attenuation
+        self._impl = impl
+
+        ran = uniform_discr_frompartition(geometry.partition)
+        super().__init__(dom, ran, linear=True)
+
+    def _call(self, x):
+        """Compute the discrete SPECT ptojection
+
+        Parameters
+        ----------
+        x : `DiscreteLpVector`
+           Element in the domain of the operator to be forward projected
+
+        Returns
+        -------
+        out : `DiscreteLpVector`
+            Returns an element in the projection space
```

Comment:
as above, same [here](https://github.com/odlgroup/odl/blob/master/odl/tomo/operators/ray_trafo.py#L177), but I'll try to be more informative